xwiki-rendering-macros/xwiki-rendering-macro-message/src/main/java/org/xwiki/rendering/internal/macro/message/AbstractMessageMacro.java

```diff
@@ -62,28 +53,38 @@
     public AbstractMessageMacro(String macroName, String macroDescription)
     {
         super(macroName, macroDescription,
-            new DefaultContentDescriptor("Content of the message", true, Block.LIST_BLOCK_TYPE));
+            new DefaultContentDescriptor("Content of the message", true, Block.LIST_BLOCK_TYPE),
+            BoxMacroParameters.class);
     }
 
     @Override
-    public List<Block> execute(Object parameters, String content, MacroTransformationContext context)
-        throws MacroExecutionException
+    public List<Block> execute(BoxMacroParameters parameters, String content,
+        MacroTransformationContext context) throws MacroExecutionException
     {
         if (StringUtils.isEmpty(content)) {
```

Comment:
Might make more sense to move this check in parseContent, WDYT ?

Reply:
Actually parseContent is never called if the content is null, according to AbstractBoxMacro#execute. So if we want to keep the same behaviour as before, we might prefer to keep it there.

Reply:
OK this is a weird behavior from AbstractBoxMacro, at the very least it take into account ContentDescriptor#isMandatory() and fail instead of just do nothing if true.

Reply:
so you suggest I change the behaviour of AbstractBoxMacro to check if the content is mandatory and to move the exception we were using in message directly in AbstractBoxMacro when the content is empty and mandatory?

Reply:
I just checked DefaultBoxMacro and noticed that the content is defined as mandatory there too but never taken into account :)